### PR TITLE
Add reactor

### LINKED ISSUE #476: [FEATURE] Create a PrimaryHX
### What is your precise need ?

Being able to model heat exchange with the core on a BWR (boiling water reactor)

### What solution do you see ?

Create a PrimaryHX with current equations from SG component. Try to extend SG from this new component (but a bit tricky because will need to add the blowdown)

### Do you see any other alternative ?
Or just create another component and duplicate equations if solution above, or any more satisfactory one, does not work

### PR DESCRIPTION
**Pull request by moritzn-met**
## Goal

Add reactor component. (Fix #476) Its an additional component, I didnt manage to extend this to use for Steam Generator. (BC of different design, and purge)

## Type of change 

- [ ] Bugfix
- [x] New feature
- [ ] Refactoring change
- [ ] Release & Version Update (don't forget to change the version number in `package.mo`)

Will it break anything in previous models ?

- [ ] Breaking change (If yes, make sure to point it out in the changelog)
- [ ] Non-Breaking change

## Checklist

- [ ] I have added the appropriate tags, reviewers, projects (and detailed the size and priority of my PR) and linked issues to this PR
- [ ] I have performed a self-review of my own code
- [ ] I have checked that all existing tests pass.
- [ ] I have added/updated tests that prove my development works and does not break anything.
- [ ] I have made corresponding changes or additions to the documentation (in [Notion documentation](https://www.notion.so/metroscope/Metroscope-Modeling-Library-Documentation-MML3-WIP-50c8703c294446059d3b4a70d6ae4a71))
- [ ] I have added corresponding entries to the [Changelog](../CHANGELOG.md)
- [ ] I have checked for conflicts with target branch, and merged/rebased in consequence

You can also fill these out after creating the PR, but make sure to **check them all before submitting your PR for review**.

### CODE REVIEW

**Comment by hadrienp-met** [comment]
I think you forgot to add the test models to your commit, the "Add tests" has no changes to show

**Comment by moritzn-met** [comment]
> I think you forgot to add the test models to your commit, the "Add tests" has no changes to show

Ups yeah something went wrong. I added the tests!

File: MetroscopeModelingLibrary/WaterSteam/HeatExchangers/Reactor.mo [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  BoundaryConditions.Sink feedwater_sink annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=180,
+        origin={2,0})));
+  Power.Connectors.Inlet C_thermal_power annotation (Placement(transformation(extent={{-40,-10},{-20,10}}), iconTransformation(extent={{-40,-10},{-20,10}})));
+equation
+
+  // Fault modes
+  if not faulty then
+    feed_water_flow_rate_measurement_bias = 0; // a bias of +5 kg/s means that 1005 kg/s is measured, while only 1000 kg/s actually go into the SG
```
Comment (hadrienp-met) [inline MetroscopeModelingLibrary/WaterSteam/HeatExchangers/Reactor.mo:40]:
```suggestion
    feed_water_flow_rate_measurement_bias = 0; // a bias of +5 kg/s means that 1005 kg/s is measured, while only 1000 kg/s actually go into the reactor

```

File: MetroscopeModelingLibrary/WaterSteam/HeatExchangers/Reactor.mo [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  end if;
+
+  // Steam mass fraction
+  steam_source.P_out = steam_pressure;
+  h_liq_sat= Water.bubbleEnthalpy(Water.setSat_p(steam_pressure));
+  h_vap_sat= Water.dewEnthalpy(Water.setSat_p(steam_pressure));
+  steam_source.h_out = vapor_fraction*h_vap_sat + (1 - vapor_fraction)*h_liq_sat;
+
+  // Mass flow rates definitions
+  Q_feedwater_measured = feedwater_sink.Q_in + feed_water_flow_rate_measurement_bias; // Note : feedwater_sink.Q_in is the actual flow going into the steam generator
```
Comment (hadrienp-met) [inline MetroscopeModelingLibrary/WaterSteam/HeatExchangers/Reactor.mo:50]:
```suggestion
  Q_feedwater_measured = feedwater_sink.Q_in + feed_water_flow_rate_measurement_bias; // Note : feedwater_sink.Q_in is the actual flow going into the reactor

```

File: MetroscopeModelingLibrary/WaterSteam/HeatExchangers/Reactor.mo [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  Q_steam = - steam_source.Q_out;
+
+  // Mass balance
+  Q_steam = Q_feedwater_measured;
+
+  // Power
+  thermal_power = Q_steam*steam_source.h_out - Q_feedwater_measured*feedwater_sink.h_in; // thermal power here refers to the estimation of thermal power so it corresponds to the measured feedwater flow rate
```
Comment (hadrienp-met) [inline MetroscopeModelingLibrary/WaterSteam/HeatExchangers/Reactor.mo:57]: We can simplify
```suggestion
  thermal_power = Q_feedwater_measured * (steam_source.h_out - feedwater_sink.h_in); // thermal power here refers to the estimation of thermal power so it corresponds to the measured feedwater flow rate

```

File: MetroscopeModelingLibrary/WaterSteam/HeatExchangers/Reactor.mo [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  h_liq_sat= Water.bubbleEnthalpy(Water.setSat_p(steam_pressure));
+  h_vap_sat= Water.dewEnthalpy(Water.setSat_p(steam_pressure));
+  steam_source.h_out = vapor_fraction*h_vap_sat + (1 - vapor_fraction)*h_liq_sat;
+
+  // Mass flow rates definitions
+  Q_feedwater_measured = feedwater_sink.Q_in + feed_water_flow_rate_measurement_bias; // Note : feedwater_sink.Q_in is the actual flow going into the steam generator
+  Q_steam = - steam_source.Q_out;
+
+  // Mass balance
+  Q_steam = Q_feedwater_measured;
```
Comment (hadrienp-met) [inline MetroscopeModelingLibrary/WaterSteam/HeatExchangers/Reactor.mo:54]: Do we still need `Q_steam `? as it is equal to `Q_feedwater_measured`

**Review by hadrienp-met — APPROVED** [review]
A few comments, but it looks good to me

**Review by hadrienp-met — CHANGES_REQUESTED** [review]
Don't forget to update the changelog and add the component to the documentation

**Review by hadrienp-met — CHANGES_REQUESTED** [review]
You need to update the changelog

**Review by hadrienp-met — APPROVED** [review]

**Pull request merged**